# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1530122434 14400
#      Wed Jun 27 14:00:34 2018 -0400
# Node ID 49d3c6344afe6111e4fb5f9d030093634fe36e69
# Parent  2710440ea566c423ee5e831708280ba11386dfd6
mkoctfile: fix one more logic error

* mkoctfile.in.cc (main): If compile_only, then DO exit after all
compile commands are done and before the link commands.

diff --git a/src/mkoctfile.in.cc b/src/mkoctfile.in.cc
--- a/src/mkoctfile.in.cc
+++ b/src/mkoctfile.in.cc
@@ -987,16 +987,21 @@ main (int argc, char **argv)
       else
         {
           std::cerr << "mkoctfile: no way to compile C++ file " << f
                     << std::endl;
           return 1;
         }
     }
 
+  // If we are only compliling, we are done.
+
+  if (compile_only)
+    return 0;
+
   if (objfiles.empty ())
     {
       std::cerr << "mkoctfile: no objects to link" << std::endl;
       return 1;
     }
 
   std::string octave_libs;
 #if defined (OCTAVE_USE_WINDOWS_API) || defined(CROSS)
